### PR TITLE
fix(block): saturate block edge arithmetic

### PR DESCRIPTION
**Pull request by joshka**
### Motivation

- Block border and title layout used unchecked `u16` arithmetic in several places.
- In debug builds that can panic on tiny or edge-case geometry; in release builds the same arithmetic wraps.
- The original report came from merge-border rendering on tiny areas, but the same pattern appeared in title layout and spacing helpers as well.

### Description

- Use saturating arithmetic in `Block::inner`, `render_sides`, `render_corners`, `titles_area`, and `vertical_space`.
- Clamp rendered title widths to `u16` for layout arithmetic.
- Replace unchecked title-width accumulation and cursor-advance math with bounded arithmetic.
- Add debug-only regression tests covering empty areas, maximal padding, title-area edge cases, and very large title widths.

### Testing

- Ran `cargo test -p ratatui-widgets block::tests`.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_69dbd35ed38c833094a9fa56b239f0d0)

### CODE REVIEW

File: ratatui-widgets/src/block.rs [hunk 1]
```diff
@@ -806,8 +806,10 @@ impl Block<'_> {
         let is_replace = self.merge_borders != MergeStrategy::Replace;
         let left_inset = left + u16::from(is_replace && self.borders.contains(Borders::LEFT));
         let top_inset = top + u16::from(is_replace && self.borders.contains(Borders::TOP));
```
Comment (joshka) [inline ratatui-widgets/src/block.rs]: I'm not sure if these additions would possibly also have overflows - need to check

File: ratatui-widgets/src/block.rs [hunk 2]
```diff
@@ -1040,16 +1050,21 @@ impl Block<'_> {
             .map(|(_, line)| line)
     }
 
+    /// Return the rendered line width clamped to `u16` for layout arithmetic.
+    fn line_width_u16(line: &Line<'_>) -> u16 {
+        line.width().min(u16::MAX as usize) as u16
+    }
```
Comment (orhun) [inline ratatui-widgets/src/block.rs:1056]: I assume we are using `CellWidth` here, would it maybe make sense to add a new trait function such as `width_clamped` instead?

Comment (joshka) [reply ratatui-widgets/src/block.rs:1056]: I think it's probably not worth the hassle IMO. I haven't seen this needed in other places yet. (YAGNI)

Comment (orhun) [reply ratatui-widgets/src/block.rs:1056]: fair enough

**Review by orhun — APPROVED** [review]
